tsl/src/reorder.c

```diff
@@ -101,12 +104,59 @@ tsl_reorder_chunk(PG_FUNCTION_ARGS)
 	if (!OidIsValid(wait_id))
 		PreventInTransactionBlock(true, "reorder");
 
-	reorder_chunk(chunk_id, index_id, verbose, wait_id);
+	reorder_chunk(chunk_id, index_id, verbose, wait_id, InvalidOid, InvalidOid);
+	PG_RETURN_VOID();
+}
+
+Datum
+tsl_move_chunk(PG_FUNCTION_ARGS)
+{
+	Oid chunk_id = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
```

Comment:
should we check that the heap and indexes are not already in the right namespace? Not sure here because as it is now this will still do a reorder, which may be useful.

Reply:
It's unclear to me which behavior would be proper, given that this does also reorder...given that we'd have to loop through all indexes, and then we'd not just move that index, but actually still do the reorder, it seems like it'd be harder to understand than just saying, if you move a table to the tablespace it's already in, it gets reordered and re-written in that tablespace, so don't do it willy-nilly.